fe/fe-core/src/main/java/com/starrocks/planner/IcebergDeleteSink.java

```diff
@@ -0,0 +1,158 @@
+import com.starrocks.credential.CloudConfiguration;
+import com.starrocks.qe.SessionVariable;
+import com.starrocks.thrift.TCompressionType;
+import com.starrocks.thrift.TDataSink;
+import com.starrocks.thrift.TDataSinkType;
+import com.starrocks.thrift.TExplainLevel;
+import com.starrocks.thrift.TIcebergTableSink;
+import com.starrocks.type.IntegerType;
+import com.starrocks.type.VarcharType;
+import org.apache.iceberg.Table;
+
+import static com.starrocks.sql.ast.OutFileClause.PARQUET_COMPRESSION_TYPE_MAP;
+
+/**
+ * IcebergDeleteSink is used to support delete operations to Iceberg tables.
+ * It supports write position delete files
+ * <p>
+ * Required columns:
+ * - _file (STRING): Path of the data file
+ * - _pos (BIGINT): Row position within the file
+ */
+public class IcebergDeleteSink extends DataSink {
+    protected final TupleDescriptor desc;
+    private IcebergTable icebergTable;
+    private final long targetTableId;
+    private final String tableLocation;
+    private final String dataLocation;
+    private final String compressionType;
+    private final long targetMaxFileSize;
+    private final String tableIdentifier;
+    private CloudConfiguration cloudConfiguration;
+
+    /**
+     * Constructor for IcebergDeleteSink
+     *
+     * @param icebergTable    The target Iceberg table
+     * @param desc            Tuple descriptor containing operation columns
+     * @param sessionVariable Session variables for configuration
+     */
+    public IcebergDeleteSink(IcebergTable icebergTable, TupleDescriptor desc,
+                             SessionVariable sessionVariable) {
+        this.icebergTable = icebergTable;
+        Table nativeTable = icebergTable.getNativeTable();
+        this.desc = desc;
+        this.tableLocation = nativeTable.location();
+        this.dataLocation = IcebergUtil.tableDataLocation(nativeTable);
+        this.targetTableId = icebergTable.getId();
+        this.tableIdentifier = icebergTable.getUUID();
+        this.compressionType = sessionVariable.getConnectorSinkCompressionCodec();
+        this.targetMaxFileSize = sessionVariable.getConnectorSinkTargetMaxFileSize() > 0 ?
+                sessionVariable.getConnectorSinkTargetMaxFileSize() : 1024L * 1024 * 1024;
+    }
+
+    public void init() {
+        String catalogName = icebergTable.getCatalogName();
+        this.cloudConfiguration = IcebergUtil.getVendedCloudConfiguration(catalogName, icebergTable);
+        // Validate tuple descriptor contains required columns
+        validateDeleteTuple(desc);
+    }
+
+    /**
+     * Validate that the tuple descriptor contains required columns
+     *
+     * @param desc The tuple descriptor to validate
+     */
+    private void validateDeleteTuple(TupleDescriptor desc) {
+        boolean hasFilePathColumn = false;
+        boolean hasPosColumn = false;
+
+        for (SlotDescriptor slot : desc.getSlots()) {
+            if (slot.getColumn() != null) {
```

Comment:
```suggestion
            if (slot.getColumn() == null) {
                continue;
            }
```
In this way, it will make the code more readable, because the indentation will be less.